src/plugins/ui_stateful/selection.ts

```diff
@@ -580,15 +580,26 @@ export class GridSelectionPlugin extends UIPlugin {
 
     const toRemove = isBasedBefore ? cmd.elements.map((el) => el + thickness) : cmd.elements;
     let currentIndex = cmd.base;
+
+    const resizingGroups: Record<number, number[]> = {};
+
     for (const element of toRemove) {
       const size = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, element);
+      const currentSize = this.getters.getHeaderSize(cmd.sheetId, cmd.dimension, currentIndex);
+      if (size != currentSize) {
+        resizingGroups[size] ??= [];
+        resizingGroups[size].push(currentIndex);
+        currentIndex += 1;
+      }
+    }
+
+    for (const size in resizingGroups) {
```

Comment:
Maybe we could add a test with `spyDispatch` to ensure the elements are grouped ?